test/test_autograd.py

```diff
@@ -4649,14 +4642,15 @@ def hook(t_):
         ):
             torch._C._current_graph_task_execution_order()
 
-        # Errors when context manager not enabled
-        t = torch.tensor(1.0, requires_grad=True).clone().sin().exp()
-        all_hooks.append(t.register_hook(hook))
-        with self.assertRaisesRegex(
-            RuntimeError,
-            "expects the current backward to be executed with multithreading disabled",
-        ):
-            t.backward()
+        # Errors when context manager not enabled and multithreading happens
```

Comment:
I'm surprised here. For cpu-only tasks, didn't we already run in a singlethread?

Reply:
Yes, but we didn't bother checking and naively always error when the multithreading flag is not explicitly set! :P